conans/client/conf/detect.py

```diff
@@ -235,9 +235,25 @@ def detected_os():
         return "Macos"
     if result.startswith("CYGWIN"):
         return "Windows"
+    if result.startswith("MINGW32_NT") or result.startswith("MINGW64_NT"):
+        return "Windows"
+    if result.startswith("MSYS_NT"):
+        return "Windows"
     return result
 
 
+def detect_subsystem():
```

Comment:
I'm wondering if we should improve https://github.com/conan-io/conan/blob/760a071260bc1c3b777356a7e4f9c639a5cd3eaa/conans/client/tools/oss.py#L263.
to also detect when you are actually running conan inside the subsystem, and not only having a bash available.
From the recipes point of view if I ask `self.os_info.detect_windows_subsystem()` if should return the subsystem I'm running. It will increase the compatibility of the recipes with Conan running inside the container.
But I'm not saying that we should use the `detect_windows_subsystem` without changes in this detection, because here is important to distinguis that we are running INSIDE the subsystem.

But anyway, look at my following comment, I'm not sure if we should adjust the subsystem subsetting even detecting we are running inside a subsystem.